Index: configure
===================================================================
--- configure	(revision 32642)
+++ configure	(working copy)
@@ -97,6 +97,7 @@
 add_engine parallaction "Parallaction" yes
 add_engine queen "Flight of the Amazon Queen" yes
 add_engine saga "SAGA" yes
+add_engine sci "SCI" no
 add_engine sky "Beneath a Steel Sky" yes
 add_engine sword1 "Broken Sword 1" yes
 add_engine sword2 "Broken Sword 2" yes
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 32642)
+++ engines/engines.mk	(working copy)
@@ -82,6 +82,11 @@
 MODULES += engines/saga
 endif
 
+ifdef ENABLE_SCI
+DEFINES += -DENABLE_SCI=$(ENABLE_SCI)
+MODULES += engines/sci
+endif
+
 ifdef ENABLE_SKY
 DEFINES += -DENABLE_SKY=$(ENABLE_SKY)
 MODULES += engines/sky
Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 32642)
+++ base/plugins.cpp	(working copy)
@@ -131,6 +131,9 @@
 		#if PLUGIN_ENABLED_STATIC(SAGA)
 		LINK_PLUGIN(SAGA)
 		#endif
+		#if PLUGIN_ENABLED_STATIC(SCI)
+		LINK_PLUGIN(SCI)
+		#endif
 		#if PLUGIN_ENABLED_STATIC(SKY)
 		LINK_PLUGIN(SKY)
 		#endif
Index: Makefile
===================================================================
--- Makefile	(revision 32642)
+++ Makefile	(working copy)
@@ -9,7 +9,7 @@
 
 DEFINES     := -DHAVE_CONFIG_H
 LDFLAGS     :=
-INCLUDES    := -I. -I$(srcdir) -I$(srcdir)/engines
+INCLUDES    := -I. -I$(srcdir) #-I$(srcdir)/engines
 LIBS        :=
 OBJS        :=
 DEPDIR      := .deps
