From 229c16de0b827321a0c3e55975e980017d234d43 Mon Sep 17 00:00:00 2001
From: Charlie Birks <charlie@daft.games>
Date: Tue, 22 Mar 2022 13:13:34 +0000
Subject: [PATCH 1/2] Update SDL2 for #16462

---
 tools/ports/sdl2.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/ports/sdl2.py b/tools/ports/sdl2.py
index f499ca8fa15..7c0a30cbd71 100644
--- a/tools/ports/sdl2.py
+++ b/tools/ports/sdl2.py
@@ -5,8 +5,8 @@
 
 import os
 
-TAG = 'release-2.0.20'
-HASH = '67e1abe1183b04836b35d724fd495c83c9559b4530d4a5c9bcc89648af0ac7cc51c02f7055a1664fe5f5f90953d22a6c431fa8bc5cdd77c94a97f107c47e2d62'
+TAG = '4b8d69a41687e5f6f4b05f7fd9804dd9fcac0347'
+HASH = '2d4d577c7584da22306b05a44bc08200460a33cd414fed2dc948e2a86e7b2d1a5cbc13bacadb63618823ba63c210f21c570adbab39f7645bf902196fa91c6b4e'
 SUBDIR = 'SDL-' + TAG
 
 

From 1eb16caf951bf0a38dda07d3335b4fdeb397ebc7 Mon Sep 17 00:00:00 2001
From: Charlie Birks <charlie@daft.games>
Date: Fri, 25 Mar 2022 11:24:00 +0000
Subject: [PATCH 2/2] Add an extra move to the SDL2 mouse test

The "first" event now has valid relative motion, so don't need that workaround either.
---
 tests/sdl2_mouse.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/sdl2_mouse.c b/tests/sdl2_mouse.c
index f7786cdb02d..6ec8871d427 100644
--- a/tests/sdl2_mouse.c
+++ b/tests/sdl2_mouse.c
@@ -26,11 +26,10 @@ void one() {
         printf("motion : %d,%d  %d,%d\n", m->x, m->y, m->xrel, m->yrel);
 
         if (mouse_motions == 0) {
-          // xrel/yrel will be zero for the first motion
 #ifdef TEST_SDL_MOUSE_OFFSETS
-          assert(eq(m->x, 5) && eq(m->y, 15) && eq(m->xrel, 0) && eq(m->yrel, 0));
+          assert(eq(m->x, 5) && eq(m->y, 15) && eq(m->xrel, 5) && eq(m->yrel, 15));
 #else
-          assert(eq(m->x, 10) && eq(m->y, 20) && eq(m->xrel, 0) && eq(m->yrel, 0));
+          assert(eq(m->x, 10) && eq(m->y, 20) && eq(m->xrel, 10) && eq(m->yrel, 20));
 #endif
         } else if (mouse_motions == 1) {
 #ifdef TEST_SDL_MOUSE_OFFSETS
@@ -93,6 +92,7 @@ int main() {
 }
 
 void main_2(void* arg) {
+  emscripten_run_script("window.simulateMouseEvent(0, 0, -1)");
   emscripten_run_script("window.simulateMouseEvent(10, 20, -1)"); // move from 0,0 to 10,20
   emscripten_run_script("window.simulateMouseEvent(10, 20, 0)"); // click
   emscripten_run_script("window.simulateMouseEvent(10, 20, 0)"); // click some more, but this one should be ignored through PeepEvent
